R-packages/evalcast/R/target.R

```diff
@@ -23,85 +22,100 @@ get_target_period <- function(forecast_date, incidence_period, ahead) {
   assert_that(incidence_period == "epiweek",
               msg="Unsupported `incidence_period`.")
   # incidence_period: epiweek
-  ew_frcst_date <- MMWRweek(forecast_date) # get epiweek of forecast_dates
-  sunday_of_ew_frcst_date <- MMWRweek2Date(MMWRyear = ew_frcst_date$MMWRyear,
-                                           MMWRweek = ew_frcst_date$MMWRweek,
-                                           MMWRday = 1) # 1 is Sunday
+  ew_frcst_date <- MMWRweek::MMWRweek(forecast_date) # get epiweek of forecast_dates
+  sunday_of_ew_frcst_date <- MMWRweek::MMWRweek2Date(
+    MMWRyear = ew_frcst_date$MMWRyear,
+    MMWRweek = ew_frcst_date$MMWRweek,
+    MMWRday = 1) # 1 is Sunday
   # From https://github.com/reichlab/covid19-forecast-hub/blob/master/data-processed/README.md:
   # "For week-ahead forecasts with forecast_date of Sunday or Monday of EW12, a
   # 1 week ahead forecast corresponds to EW12 and should have target_end_date of
   # the Saturday of EW12. For week-ahead forecasts with forecast_date of Tuesday
   # through Saturday of EW12, a 1 week ahead forecast corresponds to EW13 and
   # should have target_end_date of the Saturday of EW13."
-  week_ahead <- ifelse(wday(forecast_date) <= 2, # forecasting on a Sun/Monday
+  week_ahead <- ifelse(lubridate::wday(forecast_date) <= 2, # forecasting on a Sun/Monday
```

Comment:
lines 7,8 technically make it unnecessary. My preference is: (1) use the namespace qualification if it's a handful of functions from some package or (2) import the whole package and drop the namespaces. I like 1 in case I don't know where the function came from later. I would generally avoid (2) but dplyr/ggplot2 are so ubiquitous it's really annoying. The @importFrom tags let you drop the namespace for a single imported function, but I don't like doing that.

So in this case I removed 7,8.